--- ./frameworks/native/opengl/libs/EGL/eglApi.cpp	2016-12-08 15:08:29.737260595 +0000
+++ ./frameworks/native/opengl/libs/EGL/eglApi.cpp2	2016-12-09 14:09:06.617660961 +0000
@@ -481,15 +481,15 @@
         // modify the EGLconfig's format before setting the native window's
         // format.
 
-        // by default, just pick RGBA_8888
-        EGLint format = HAL_PIXEL_FORMAT_RGBA_8888;
+        // by default, just pick BGRA_8888
+        EGLint format = HAL_PIXEL_FORMAT_BGRA_8888;
         android_dataspace dataSpace = HAL_DATASPACE_UNKNOWN;
 
         EGLint a = 0;
         cnx->egl.eglGetConfigAttrib(iDpy, config, EGL_ALPHA_SIZE, &a);
         if (a > 0) {
             // alpha-channel requested, there's really only one suitable format
-            format = HAL_PIXEL_FORMAT_RGBA_8888;
+            format = HAL_PIXEL_FORMAT_BGRA_8888;
         } else {
             EGLint r, g, b;
             r = g = b = 0;
@@ -1859,7 +1859,7 @@
     // Validate format.
     if (red_size == 8 && green_size == 8 && blue_size == 8) {
         if (alpha_size == 8) {
-            format = HAL_PIXEL_FORMAT_RGBA_8888;
+            format = HAL_PIXEL_FORMAT_BGRA_8888;
         } else {
             format = HAL_PIXEL_FORMAT_RGB_888;
         }
