--- frameworks/base/core/java/android/view/GestureDetector.java	2016-12-08 15:07:53.705260420 +0000
+++ frameworks/base/core/java/android/view/GestureDetector.java2	2016-12-09 13:57:25.521657573 +0000
@@ -488,6 +488,15 @@
      *              else false.
      */
     public boolean onTouchEvent(MotionEvent ev) {
+        switch(ev.getAction() & MotionEvent.ACTION_MASK) {
+        case MotionEvent.ACTION_HOVER_MOVE:
+        case MotionEvent.ACTION_HOVER_ENTER:
+        case MotionEvent.ACTION_HOVER_EXIT:
+        case MotionEvent.ACTION_BUTTON_PRESS:
+        case MotionEvent.ACTION_BUTTON_RELEASE:
+        case MotionEvent.ACTION_SCROLL:
+            return false;
+        }
         if (mInputEventConsistencyVerifier != null) {
             mInputEventConsistencyVerifier.onTouchEvent(ev, 0);
         }
