--- frameworks/base/core/java/android/view/GestureDetector.java 2015-12-31 14:17:22.342089597 +0000
+++ frameworks/base/core/java/android/view/GestureDetector.java2        2015-12-31 17:00:12.618371748 +0000
@@ -488,7 +488,16 @@
      *              else false.
      */
     public boolean onTouchEvent(MotionEvent ev) {
-        if (mInputEventConsistencyVerifier != null) {
+      switch(ev.getAction() & MotionEvent.ACTION_MASK) {
+         case MotionEvent.ACTION_HOVER_MOVE:
+         case MotionEvent.ACTION_HOVER_ENTER:
+         case MotionEvent.ACTION_HOVER_EXIT:
+         case MotionEvent.ACTION_BUTTON_PRESS:
+         case MotionEvent.ACTION_BUTTON_RELEASE:
+         case MotionEvent.ACTION_SCROLL:
+             return false;
+         }    
+      if (mInputEventConsistencyVerifier != null) {
             mInputEventConsistencyVerifier.onTouchEvent(ev, 0);
         }
